scalyr_agent/tests/scalyr_logging_test.py

```diff
@@ -240,14 +240,20 @@ def test_rate_limit_no_write_rate(self):
             use_disk=True,
             logs_directory=os.path.dirname(self.__log_path),
             agent_log_file_path=self.__log_path,
-            max_write_burst=250,
+            max_write_burst=500,
```

Comment:
NOTE: Using this static value here still isn't the best / most robust approach since test may start to fail if we change the default format string and / or the warning to be longer.

An approach to make this more robust and future proof is to instantiate two ``LogRecord`` objects with the corresponding messages, obtain the length of the fully formatted message (for the second log message, this includes the warning) and then make sure that ``max_write_burst = (len(message_one_formatted) + len(message_two_formatted_with_warning)) + 1`` or something along those lines.

Reply:
I'll throw out a few alternatives as well.  We could add a parameter in `AgentLogManager.set_log_destination` allowing us to disable using a formatter.  The charged bytes would be equal to just the length of the original string, making it easier to reason about the tests.  This would require a little bit of plumbing and adds in a mode that we would never really use in production.. so not perfect.

Another alternative:  Record the len of the string in the LogRecord (regardless of whether or not it keeps filtered out) and use the `last_record` to get the last record and do a check (before we do a second that the log contains `Second message`) that the length of the string is less than what the static value was.  Add a comment there to say if the length has changed, we probably need to up the static `max_write_burst`.

I wasn't clear from your approach how you would instantiate the `LogRecord` objects such that we are sure they are using the same formatter as what is used by the filter.

Reply:
I agree about the first approach - it's similar to mocking.

You need to be careful and find a good balance, because if you end up mocking too much, you may just end up testing mocks itself and not the code which is supposed to be tested.

Because of that I went with the variation of the second approach you suggested - 58ba694d1431b37830466261cf4bb2e42db77037.

I think it's reasonable enough and will make sure tests now fail fast, consistently and loudly in case a format string changes, test contains invalid value for ``max_write_burst`` or similar which is much better than intermediate failures which are hard to track down.

Storing those two values on the original ``LogRecord`` object does mean the object itself will be slightly larger, but I don't think that's a big deal since those values are very small compared to the actual log messages :)

And If I added a check to do this only during the debug / test mode, this would introduce additional code complexity for a little good reason (if that ever becomes an issue, which I doubt, we can change / improve that in the future).